key/key.go

```diff
@@ -44,9 +44,9 @@ type keyImpl struct {
 	key interface{}
 }
 
-// NewKey returns a Key object implementing the Key interface
+// New returns a Key object implementing the Key interface
```

Comment:
Yeah the original comment was odd.  I went with just `// New wraps the given value in a Key.`, how's that?